### ISSUE TITLE
Incompatible with Gradle 6

### ISSUE THREAD

**Issue by int128** [int128/gradle-swagger-generator-plugin#185]
## Describe the bug

```
Execution failed for task ':acceptance-test:validatePlugins'.
> Plugin validation failed. See https://docs.gradle.org/6.0.1/userguide/more_about_tasks.html#sec:task_input_output_annotations for more information on how to annotate task properties.
   > Warning: Type 'GenerateReDoc': property 'inputFile' is declared without normalization specified. Properties of cacheable work must declare their normalization via @PathSensitive, @Classpath or @CompileClasspath. Defaulting to PathSensitivity.ABSOLUTE.
   > Warning: Type 'GenerateReDoc': property 'wipeOutputDir' @Input properties with primitive type 'boolean' cannot be @Optional.
   > Warning: Type 'GenerateSwaggerCode': property 'adaptorFactory' is not annotated with an input or output annotation.
   > Warning: Type 'GenerateSwaggerCode': property 'configFile' is declared without normalization specified. Properties of cacheable work must declare their normalization via @PathSensitive, @Classpath or @CompileClasspath. Defaulting to PathSensitivity.ABSOLUTE.
   > Warning: Type 'GenerateSwaggerCode': property 'inputFile' is declared without normalization specified. Properties of cacheable work must declare their normalization via @PathSensitive, @Classpath or @CompileClasspath. Defaulting to PathSensitivity.ABSOLUTE.
   > Warning: Type 'GenerateSwaggerCode': property 'templateDir' is declared without normalization specified. Properties of cacheable work must declare their normalization via @PathSensitive, @Classpath or @CompileClasspath. Defaulting to PathSensitivity.ABSOLUTE.
   > Warning: Type 'GenerateSwaggerCode': property 'wipeOutputDir' @Input properties with primitive type 'boolean' cannot be @Optional.
   > Warning: Type 'GenerateSwaggerCodeHelp': property 'adaptorFactory' is not annotated with an input or output annotation.
   > Warning: Type 'GenerateSwaggerUI': property 'inputFile' is declared without normalization specified. Properties of cacheable work must declare their normalization via @PathSensitive, @Classpath or @CompileClasspath. Defaulting to PathSensitivity.ABSOLUTE.
   > Warning: Type 'GenerateSwaggerUI': property 'wipeOutputDir' @Input properties with primitive type 'boolean' cannot be @Optional.
   > Warning: Type 'ValidateSwagger': property 'inputFile' is declared without normalization specified. Properties of cacheable work must declare their normalization via @PathSensitive, @Classpath or @CompileClasspath. Defaulting to PathSensitivity.ABSOLUTE.
```

To Reproduce:
https://circleci.com/gh/int128/gradle-swagger-generator-plugin/317

Expected behavior:
It should work on Gradle 6.

## Environment
Plugin version: latest
Swagger Codegen version: latest
Gradle version: 6.0.1
Java version: 8
OS: Linux

### Build errors

- Type `GenerateReDoc`:
  - [ ] property `inputFile` is declared without normalization specified. Properties of cacheable work must declare their normalization via `@PathSensitive`, `@Classpath` or `@CompileClasspath`. Defaulting to PathSensitivity.ABSOLUTE.
  - [ ] property `wipeOutputDir` `@Input` properties with primitive type `boolean` cannot be `@Optional`.
- Type `GenerateSwaggerCode`:
  - [ ] property `adaptorFactory` is not annotated with an input or output annotation.
  - [ ] property `configFile` is declared without normalization specified. Properties of cacheable work must declare their normalization via `@PathSensitive`, `@Classpath` or `@CompileClasspath`. Defaulting to PathSensitivity.ABSOLUTE.
  - [ ] property `inputFile` is declared without normalization specified. Properties of cacheable work must declare their normalization via `@PathSensitive`, `@Classpath` or `@CompileClasspath`. Defaulting to PathSensitivity.ABSOLUTE.
  - [ ] property `templateDir` is declared without normalization specified. Properties of cacheable work must declare their normalization via `@PathSensitive`, `@Classpath` or `@CompileClasspath`. Defaulting to PathSensitivity.ABSOLUTE.
  - [ ] property `wipeOutputDir` `@Input` properties with primitive type `boolean` cannot be `@Optional`.
- Type `GenerateSwaggerCodeHelp`:
  - [ ] property `adaptorFactory` is not annotated with an input or output annotation.
- Type `GenerateSwaggerUI`:
  - [ ] property `inputFile` is declared without normalization specified. Properties of cacheable work must declare their normalization via `@PathSensitive`, `@Classpath` or `@CompileClasspath`. Defaulting to PathSensitivity.ABSOLUTE.
  - [ ] property `wipeOutputDir` `@Input` properties with primitive type `boolean` cannot be `@Optional`.
- Type `ValidateSwagger`:
  - [ ] property `inputFile` is declared without normalization specified. Properties of cacheable work must declare their normalization via `@PathSensitive`, `@Classpath` or `@CompileClasspath`. Defaulting to PathSensitivity.ABSOLUTE.

### Warnings

- [ ] The compile configuration has been deprecated for dependency declaration. This will fail with an error in Gradle 7.0. Please use the implementation configuration instead. Consult the upgrading guide for further information: https://docs.gradle.org/6.2/userguide/upgrading_version_5.html#dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations
        at build.gradle:16
- [ ] The testCompile configuration has been deprecated for dependency declaration. This will fail with an error in Gradle 7.0. Please use the testImplementation configuration instead. Consult the upgrading guide for further information: https://docs.gradle.org/6.2/userguide/upgrading_version_5.html#dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations
        at build.gradle:21

**Comment by wwadge** [comment]
Any update on this? it will start failing on gradle v7

**Comment by int128** [comment]
Fixed in 2.18.2.
